pkg/cmd/pulumi/packagecmd/package_add.go

```diff
@@ -116,7 +116,11 @@ extension, Pulumi package schema is read from it directly:
 				return fmt.Errorf("failed to create directory for SDK: %w", err)
 			}
 
-			out = filepath.Join(out, pkg.Name)
+			outName := pkg.Name
+			if pkg.Namespace != "" {
+				outName = pkg.Namespace + "_" + outName
```

Comment:
Another slight change in behaviour here, if there is a namespace set we should also disambiguate the directory based on that. Otherwise there's not gonna be enough disambiguation and we will overwrite SDKs with different namespaces.